sbtc-core/src/signer/blockchain/mod.rs

```diff
@@ -0,0 +1,46 @@
+use bitcoin::Transaction as BitcoinTransaction;
+use url::Url;
+
+use crate::{
+    signer::{PublicKeys, StacksTransaction},
+    SBTCResult,
+};
+
+/// The broker is responsible for retreiving sBTC withdrawals requests and
+/// broadcasting transactions to the stacks and bitcoin networks.
+pub struct Broker {
+    /// The bitcoin node RPC URL
+    pub bitcoin_node_rpc_url: Url,
+    /// The stacks node RPC URL
+    pub stacks_node_rpc_url: Url,
+}
+
+impl Broker {
+    /// Create a new broker
+    pub fn new(bitcoin_node_rpc_url: Url, stacks_node_rpc_url: Url) -> Self {
+        Self {
+            bitcoin_node_rpc_url,
+            stacks_node_rpc_url,
+        }
+    }
+
+    /// Retrieve the current public keys for the signers and their vote ids from the stacks node
+    pub fn public_keys(&self) -> SBTCResult<PublicKeys> {
+        todo!()
+    }
```

Comment:
Nit: More precise naming. Also, I would like to preserve the namespacing here since it contains interesting information, so instead of importing `signer::PublicKeys` we'd just use it with the namespace directly. I'm fine with the import though.

```suggestion
    pub fn signer_public_keys(&self) -> SBTCResult<PublicKeys> {
        todo!()
    }
```

Reply:
Something we should add here is reward cycle information. Either on the query or the response, but preferably on the query. I.e.

```
pub fun signer_public_keys(&self, reward_cycle) -> ...
```

That would make the function call idempotent, which would help us build a much more reliable signer.

Reply:
Or possibly not completely idempotent, since the broker might not have the information. So we'd return an Option or a Result. But the returned value if any would always be consistent.